### PR TITLE
overload manager: removing exceptions

### PR DESCRIPTION
**Pull request by alyssawilk**
Risk Level: low
Testing: updated tests
Docs Changes: n/a
Release Notes: n/a
https://github.com/envoyproxy/envoy-mobile/issues/176

### CODE REVIEW

File: source/server/instance_impl.cc [hunk 1]
```diff
@@ -11,8 +11,8 @@ void InstanceImpl::maybeCreateHeapShrinker() {
       std::make_unique<Memory::HeapShrinker>(dispatcher(), overloadManager(), *stats().rootScope());
 }
 
-std::unique_ptr<OverloadManager> InstanceImpl::createOverloadManager() {
-  return std::make_unique<OverloadManagerImpl>(
+absl::StatusOr<std::unique_ptr<OverloadManager>> InstanceImpl::createOverloadManager() {
```
Comment (KBaichoo) [inline source/server/instance_impl.cc:14]: nit: maybe we should add a alias for `std::unique_ptr<OverloadManager>` as done for other unique ptrs. Will reduce the nesting quite a bit.

File: mobile/library/common/engine_common.cc [hunk 2]
```diff
@@ -59,7 +59,7 @@ class ServerLite : public Server::InstanceBase {
 public:
   using Server::InstanceBase::InstanceBase;
   void maybeCreateHeapShrinker() override {}
-  std::unique_ptr<Envoy::Server::OverloadManager> createOverloadManager() override {
+  absl::StatusOr<std::unique_ptr<Envoy::Server::OverloadManager>> createOverloadManager() override {
```
Comment (KBaichoo) [inline mobile/library/common/engine_common.cc:62]: Are we trying to use overload manager in envoy mobile? It seems not. Maybe you can remind me the impetus behind removing exceptions here?

**Review by KBaichoo — APPROVED** [review]
lgtm besides minor comments / questions

File: source/server/instance_impl.cc (hunk 1 above)
Comment (alyssawilk) [reply source/server/instance_impl.cc:14]: I'm going to leave as is as I'm on a team of engineers who strongly dislike the FooPtr paragim =P

File: mobile/library/common/engine_common.cc (hunk 2 above)
Comment (alyssawilk) [reply mobile/library/common/engine_common.cc:62]: unfortunately as it's pulled in in e2e tests its "in scope" and it turns out it's helpful to remove exceptions overall (the number of config crashes I've found is distressingly high)

**Pull request merged**